apollo-federation/src/sources/connect/models/validation/mod.rs

```diff
@@ -476,241 +451,24 @@ fn validate_field(
         }
     }
 
-    // validate header arg
-    let http_headers_arg = http_arg
+    if let Some(headers) = http_arg
```

Comment:
Similarly, here the logic to assemble the headers probably belongs in the `http_heasers` module.